src/main/java/duke/Task.java

```diff
@@ -0,0 +1,80 @@
+package duke;
+
+import java.time.LocalDate;
+
+/**
+ * Abstract class that all Tasks inherit from.
+ */
+public abstract class Task {
+    private String description;
+    private boolean done;
+
+    /**
+     * Creates a task.
+     * @param description Description of task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.done = false;
+    }
+
+    /**
+     * Marks task as done.
+     */
+    public void markDone() {
+        this.done = true;
+    }
+
+    /**
+     * Marks task as not done.
+     */
+    public void unmarkDone() {
+        this.done = false;
+    }
+
+    /**
+     * Returns common part of all tasks for file representation
+     * @return "{1 if done else 0} | {description}"
+     */
+    public String toFileRepresentation() {
+        return String.format("%d | %s", this.done ? 1 : 0, this.description);
+    }
+
+    /**
+     * Returns nothing because abstract tasks should not be created.
+     * @param rep String of task representation
+     * @return Nothing.
+     */
+    public static Task fromFileRepresentation(String rep) {
```

Comment:
Since child classes of Task will override this method, perhaps an abstract method might be preferred?